### ISSUE TITLE
Training Mismatch between OpenRLHF and Verl

### ISSUE THREAD

**Issue by yph22** [verl-project/verl#5355]
### System Info

----------Python Info----------
Version      : 3.10.0
Compiler     : GCC 7.5.0
Build        : ('default', 'Mar  3 2022 09:58:08')
Arch         : ('64bit', 'ELF')
------------Pip Info-----------
Version      : 25.3
Directory    :/miniconda3/envs/verl-gad/lib/python3.10/site-packages/pip
vllm         : 0.11.0
sglang       : not found.
ray          : 2.48.0
torch        : 2.8.0+cu128
----------verl Info-----------
Version      : 0.7.0.dev
Directory    : /verl/verl

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

Hello, I’ve been running experiments for a week recently, trying to migrate the **RLOO** training code from **openrlhf** to **verl**, but it just isn’t working in verl 😭. In openrlhf, **both the training reward and response length showed a clear upward and converging trend, but in verl, both keep fluctuating back and forth with no upward momentum.**

The training parameters of the two frameworks (including downstream reward server sampling parameters) are basically unified. I’ve also tried adjusting some parameters within verl (like **loss_agg_mode**, **rollout_correction.bypass_mode**, etc.), but there’s almost no change—the reward just won’t go up 😭.

Could any experts help figure out what’s wrong? 🥺

Basical codes for openrlhf:
`python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 8 \
   --remote_rm_url "http://${REWARD_SERVER_IP}:${REWARD_PORT}/get_reward" \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 8 \
   --vllm_num_engines 8 \
   --vllm_tensor_parallel_size 1 \
   --colocate_all_models \
   --vllm_enable_sleep \
   --vllm_gpu_memory_utilization 0.4 \
   --enable_prefix_caching \
   --pretrain "$PRETRAIN_MODEL" \
   --save_path "$SAVE_PATH/$MODEL_CPK_NAME" \
   --remove_pi_old \
   --load_checkpoint \
   --format_weight 0.0 \
   --micro_train_batch_size 4 \
   --train_batch_size 1024 \
   --micro_rollout_batch_size 128 \
   --rollout_batch_size 128 \
   --temperature 1.0 \
   --n_caps_per_prompt 8 \
   --n_samples_per_prompt 8 \
   --max_epochs 1 \
   --num_episodes 1 \
   --prompt_max_len 4096 \
   --max_samples 100000 \
   --generate_max_len 4096 \
   --advantage_estimator rloo \
   --zero_stage 0 \
   --bf16 \
   --actor_learning_rate 1e-6 \
   --lr_warmup_steps 20 \
   --init_kl_coef 0.0 \
   --prompt_data "$DATASET" \
   --input_key message \
   --normalize_reward \
   --gradient_checkpointing \
   --save_steps 50 \
   --ckpt_path "$SAVE_PATH/$MODEL_CPK_NAME/ckpt" \
   --save_hf_ckpt \
   --train_vlm \
   --attn_implementation flash_attention_2`

For Verl:
`python -m verl.trainer.main_ppo \
    data.train_files=${DATASET} \
    data.val_files=${DATASET} \
    data.train_batch_size=128 \
    data.gen_batch_size=128 \
    data.max_prompt_length=4096 \
    data.max_response_length=4096 \
    actor_rollout_ref.model.path=${CAPTION_MODEL} \
    algorithm.adv_estimator=rloo \
    actor_rollout_ref.actor.optim.lr_scheduler_type=cosine \
    actor_rollout_ref.actor.optim.min_lr_ratio=0.1 \
    actor_rollout_ref.actor.optim.lr_warmup_steps=20 \
    actor_rollout_ref.actor.strategy=fsdp \
    actor_rollout_ref.actor.use_kl_loss=false \
    actor_rollout_ref.actor.use_dynamic_bsz=true \
    actor_rollout_ref.actor.use_remove_padding=true \
    actor_rollout_ref.actor.loss_agg_mode=seq-mean-token-mean \
    actor_rollout_ref.actor.ppo_mini_batch_size=128 \
    actor_rollout_ref.actor.ppo_max_token_len_per_gpu=32768 \
    actor_rollout_ref.rollout.name=vllm \
    actor_rollout_ref.rollout.mode=async \
    actor_rollout_ref.rollout.n=8 \
    actor_rollout_ref.rollout.prompt_length=4096 \
    actor_rollout_ref.rollout.response_length=4096 \
    actor_rollout_ref.rollout.max_model_len=8292 \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.7 \
    actor_rollout_ref.rollout.log_prob_max_token_len_per_gpu=32768 \
    actor_rollout_ref.ref.log_prob_max_token_len_per_gpu=32768 \
    actor_rollout_ref.rollout.log_prob_max_token_len_per_gpu=32768 \
    actor_rollout_ref.actor.fsdp_config.model_dtype=bf16 \
    trainer.total_epochs=1`

### Expected behavior

They should align with each other.

**Comment by yph22** [comment]
Training Curves:
**1. OpenRLHF:**

<img width="1279" height="539" alt="Image" src="https://github.com/user-attachments/assets/c82fc6d8-95ed-479b-aa97-140c4a2183af" />

<img width="1279" height="540" alt="Image" src="https://github.com/user-attachments/assets/ead36e31-50e3-4c59-a329-6258b826ace5" />

**2. Verl:**

<img width="1719" height="532" alt="Image" src="https://github.com/user-attachments/assets/b0f7bbf7-1c86-4ecf-906b-ce2ba58ed485" />

<img width="1713" height="532" alt="Image" src="https://github.com/user-attachments/assets/07a29942-95fe-4939-a903-0c9cf9a5c7db" />

**Comment by yph22** [comment]
I’m excited to share that I’ve finally solved this problem by **increasing the learning rate 10x** compared to openrlhf. In openrlhf, it was set to 1e-6, so I reset it to 1e-5 in verl. Now the training curves align with each other.

<img width="1721" height="540" alt="Image" src="https://github.com/user-attachments/assets/0591a6c2-30a6-4170-89aa-73ad8f3a9c6b" />

<img width="1713" height="532" alt="Image" src="https://github.com/user-attachments/assets/20c9529d-0320-495e-88f1-7bc826d13378" />
